src/components/ChatRoomPage/ChatItem.tsx

```diff
@@ -0,0 +1,69 @@
+import styled from "styled-components";
+import { Chat, User } from "../../common/interface";
+import { ProfileImage } from "../../common/StyledComponent";
+
+interface ChatItemProps {
+  chat: Chat;
+  isCurUser: boolean;
+  sender: User;
+}
+
+const ChatItem = ({ chat, isCurUser, sender }: ChatItemProps) => {
+  const time = String(new Date(chat.date).getHours()).padStart(2, "0");
+  const minute = String(new Date(chat.date).getMinutes()).padStart(2, "0");
+
+  return (
+    <Wrapper isCurUser={isCurUser}>
+      {isCurUser ? (
+        <>
+          <ChatWrapper>
+            {time}:{minute}
+            <ChatBalloon isCurUser={true}>{chat.text}</ChatBalloon>
+          </ChatWrapper>
+        </>
+      ) : (
+        <>
+          <ProfileImage src={sender.profileImage} />
+          <ContentWrapper>
+            {sender.name}
+            <ChatWrapper>
+              <ChatBalloon isCurUser={false}>{chat.text}</ChatBalloon>
+              {time}:{minute}
+            </ChatWrapper>
+          </ContentWrapper>
+        </>
+      )}
+    </Wrapper>
+  );
+};
+
+const Wrapper = styled.div`
+  display: flex;
+  gap: 10px;
+  justify-content: ${({ isCurUser }: { isCurUser: boolean }) =>
+    isCurUser ? "flex-end" : "flex-start"};
+  width: 100%;
+`;
```

Comment:
```suggestion
const Wrapper = styled.div<{ isCurUser?: string}>`
  justify-content: ${(props) => props.isCurUser|| "flex-start"};
`;
```
`
styled-component`에서 `props`를 보낼때 이렇게도 사용하실 수 있습니다! `boolean` 만 이용해서 사용하셨길래 또 다른 방법 놓고갑니다!

Reply:
저도` boolean`을 이용했는데,, 배워갑니다!.

Reply:
우왕.. 컴포넌트 상위에서 type을 지정할수도 있군요..!! 저도 배워갑니다👍